functests/xt_u32/xt_u32.go

```diff
@@ -0,0 +1,223 @@
+
+const xt_u32LoadPath = "/etc/modules-load.d/xt_u32-load.conf"
+
+var (
+	xt_u32NodeSelector string
+	hasNonCnfWorkers   bool
+)
+
+func init() {
+	roleWorkerCNF := os.Getenv("ROLE_WORKER_CNF")
+	if roleWorkerCNF != "" {
+		xt_u32NodeSelector = fmt.Sprintf("node-role.kubernetes.io/%s=", roleWorkerCNF)
+	}
+
+	hasNonCnfWorkers = true
+	if os.Getenv("XT_U32_TEST_HAS_NON_CNF_WORKERS") == "false" {
+		hasNonCnfWorkers = false
+	}
+}
+
+var _ = Describe("xt_u32", func() {
+	var testNode string
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(TestNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = namespaces.Clean(TestNamespace, "xt-u32", client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		if xt_u32NodeSelector == "" {
+			xt_u32NodeSelector, err = findXT_U32NodeSelector()
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	BeforeEach(func() {
+		if !hasNonCnfWorkers {
+			Skip("Skipping as no non-enabled nodes are available")
+		}
+
+		namespaces.Clean(TestNamespace, "xt-u32", client.Client)
+		By("Choosing the test node")
+		nodes, err := client.Client.Nodes().List(context.Background(), metav1.ListOptions{
+			LabelSelector: xt_u32NodeSelector,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		filtered, err := utilNodes.MatchingOptionalSelector(nodes.Items)
+		Expect(err).ToNot(HaveOccurred())
+
+		if discovery.Enabled() && len(filtered) == 0 {
+			Skip("Did not find a node without xt_u32 module enabled")
+		} else {
+			Expect(len(filtered)).To(BeNumerically(">", 0))
+		}
+		testNode = filtered[0].ObjectMeta.Labels[hostnameLabel]
+	})
+
+	Context("Validate the module is enabled and works", func() {
+		It("Should create an iptables rule inside a pod that has the module enabled", func() {
+			By("Create a pod")
+			pod := xt_u32TestPod("xt-u32", testNode, TestNamespace, []string{"sleep inf"})
+			xt_u32Pod, err := client.Client.Pods(TestNamespace).Create(context.Background(), pod, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check if the pod is running")
+			Eventually(func() k8sv1.PodPhase {
+				res, err := client.Client.Pods(TestNamespace).Get(context.Background(), xt_u32Pod.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return res.Status.Phase
+			}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+
+			By("Check that xt_u32 module is loaded")
+			cmd := []string{"lsmod"}
+			out, err := pods.ExecCommand(client.Client, *xt_u32Pod, cmd)
+			Expect(err).ToNot(HaveOccurred())
+			res := strings.Contains(out.String(), "xt_u32")
+			if !res {
+				err = errors.New("xt_u32 module is not loaded")
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("Create an iptables rule within the pod")
+			cmd = []string{"iptables", "-t", "filter", "-A", "INPUT", "-i", "eth0", "-m",
+				"u32", "--u32", "6 & 0xFF = 1 && 4 & 0x3FFF = 0 && 0 >> 22 & 0x3C @ 0 >> 24 = 8",
+				"-j", "DROP"}
+			_, err = pods.ExecCommand(client.Client, *xt_u32Pod, cmd)
```

Comment:
Can you add the output as a description in case of failure?
I think it will be helpful to triage errors.

Reply:
I'd also add a descriptive info print explaining what this line does
so it's less cryptic 
IIUC it drops icmp ping (type=8, echo request) msgs from any source:
ip protocol = 1 (ICMP) && not fragmented && icmp type = 8 (echo request)

Reply:
btw, seems like this will work on IPv4 only
I guess command itself will still work on IPv6.
(so no err will occur)